pontoon/sync/formats/compare_locales.py

```diff
@@ -64,6 +78,32 @@ def __init__(self, path, source_resource=None):
     def translations(self):
         return sorted(self.entities.values(), key=lambda e: e.order)
 
+    def save(self, locale):
+        if not self.source_resource:
+            raise SyncError(
+                'Cannot save resource {0}: No source resource given.'
+                .format(self.path)
+            )
+
+        # Parse source resource
+        self.source_resource.parser.readFile(self.source_resource.path)
```

Comment:
I need to investigate why do we need to read the reference file again. I was expecting `list(self.source_resource.parser.walk())` to JustWork™, but for some reason `self.source_resource.parser.ctx.contents` is empty.